src/rpc/rawtransaction.cpp

```diff
@@ -579,39 +580,75 @@ static RPCHelpMan decodescript()
     std::vector<std::vector<unsigned char>> solutions_data;
     const TxoutType which_type{Solver(script, solutions_data)};
 
-    if (which_type == TxoutType::SCRIPTHASH) {
-        // P2SH cannot be wrapped in a P2SH. If this script is already a P2SH,
-        // don't return the address for a P2SH of the P2SH.
+    // Check if wrapping makes sense at all
+    switch (which_type) {
+    case TxoutType::MULTISIG:
+    case TxoutType::NONSTANDARD:
+    case TxoutType::PUBKEY:
+    case TxoutType::PUBKEYHASH:
+    case TxoutType::WITNESS_V0_KEYHASH:
+    case TxoutType::WITNESS_V0_SCRIPTHASH:
+        // Can be wrapped
+        break;
+    case TxoutType::NULL_DATA:
+    case TxoutType::SCRIPTHASH:
+    case TxoutType::WITNESS_UNKNOWN:
+    case TxoutType::WITNESS_V1_TAPROOT:
+        // Should not be wrapped, so return early
         return r;
     }
+    if (!script.HasValidOps() || script.IsUnspendable()) {
+        return false;
```

Comment:
In fa200104235398a0c507122b95e0899022650be0 "rpc: Only allow specific types to be P2(W)SH wrapped in decodescript"

Returning false here is incorrect. If a script contained invalid opcodes or is otherwise unspendable, we still want to decode them, we just don't want to get it wrapped in any addresses

Any such scripts would result in an error similar to the following:

```
> src/bitcoin-cli -regtest decodescript 20393ea2d59e5565796ef02e1f91d5efe0577be63e5cf460503009c01d368c7aebbe 
error code: -1
error message:
rpc/util.cpp:577 (HandleRequest)
Internal bug detected: 'std::any_of(m_results.m_results.begin(), m_results.m_results.end(), [ret](const RPCResult& res) { return res.MatchesType(ret); })'
You may report this issue here: https://github.com/bitcoin/bitcoin/issues
```

There appears to be a separate bug(?) where `OP_SUCESSx` and `OP_CHECKSIGADD` are being caught by this condition too.

Reply:
Force pushed `return r;`.